openapi-cli/src/test/resources/ballerina-to-openapi/request_body/union_type.bal

```diff
@@ -0,0 +1,36 @@
+import ballerina/http;
+
+public type ABC record {
+    int id;
+    string name;
+};
+
+service /payloadV on new http:Listener(0) {
+    resource function post path(ABC? payload) {
+    }
+    resource function post path2(ABC|xml payload) {
```

Comment:
How about this scenario? Do we support this?
```bal
type UnionType ABC|xml;

...

resource function post path(UnionType payload) {
```

Reply:
Yeah, we need to address this scenario, its logic may complicated and we wish to address this in the separate PR. For tracking purposes, I added the tests there  https://github.com/ballerina-platform/openapi-tools/pull/1544/commits/127fc070dec352e23a9ac62a832045e41bc8ff2a.